packages/gatsby-plugin-manifest/src/safe-sharp.js

```diff
@@ -0,0 +1,137 @@
+// this is very hacky (don't hate me), but at least it's not in [JSFuck](http://www.jsfuck.com/)
+
+const childProcess = require(`child_process`)
+const semver = require(`semver`)
+
+const originalConsoleError = console.error
+const restoreConsoleError = () => {
+  console.error = originalConsoleError
+}
+
+const handleMessage = msg => {
+  if (msg.includes(`Incompatible library version: sharp.node requires`)) {
+    restoreConsoleError()
+
+    let msg = [
+      `It looks like there are multiple versions of "sharp" module installed.`,
+      `Please update packages that depend on "sharp".`,
+      ``,
+    ]
+
+    try {
+      let tmpMsg = []
+      // npm list seems to work in yarn installed projects as well
+      const { dependencies } = JSON.parse(
+        childProcess.execSync(`npm list sharp --json`, {
```

Comment:
Does this factor in a lock file? Would this be inconsistent in case there are both lock files?

Reply:
good point, it will make use of `package-lock.json` if it's available and potentially show not accurate results. If lock file doesn't exist, it will traverse `node_modules`. This makes it very hard to support, but I think maybe opting out of it if both yarn and npm lock files exist seems reasonable?